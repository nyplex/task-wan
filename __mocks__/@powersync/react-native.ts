module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
