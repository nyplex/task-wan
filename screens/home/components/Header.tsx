import Text from "@/components/Text";
import Icon from "@/components/UI/Icon";
import { HStack } from "@/components/gluestack/hstack";

const Header = () => {
  return (
    <HStack className="justify-between items-center">
      <Text size="bodyS">Saturday, Feb 20 2022</Text>
      <Icon icon="bell" />
    </HStack>
  );
};

export default Header;
