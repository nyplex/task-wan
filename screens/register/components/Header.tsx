import { useRouter } from "expo-router";
import { useSelector } from "react-redux";
import { selectAuthStatus } from "@/redux/slices/authSlice/authSelectors";
import { Box } from "@/components/gluestack/box";
import AppTitle from "@/components/UI/AppTitle";
import BackButtonIcon from "@/components/buttons/BackButtonIcon";

const Header = () => {
  const router = useRouter();
  const isLoading = useSelector(selectAuthStatus);

  return (
    <Box>
      <BackButtonIcon
        onPress={() => router.back()}
        disabled={isLoading}
      />
      <Box className="mt-4">
        <AppTitle />
      </Box>
    </Box>
  );
};

export default Header;
