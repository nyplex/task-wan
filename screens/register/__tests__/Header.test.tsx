import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Header from "../components/Header";

jest.mock("expo-router", () => ({ useRouter: () => ({ back: jest.fn() }) }));
jest.mock("react-redux", () => ({ useSelector: () => false }));
jest.mock("react-native-reanimated", () =>
  require("react-native-reanimated/mock"),
);
jest.mock("react-native-keyboard-controller", () => ({
  useReanimatedKeyboardAnimation: () => ({ progress: { value: 0 } }),
}));
jest.mock("@/components/UI/AppTitle", () => {
  const MockAppTitle = () => <></>;
  MockAppTitle.displayName = "AppTitle";
  return MockAppTitle;
});

jest.mock("@/components/buttons/BackButtonIcon", () => {
  const TouchableOpacity = require("react-native").TouchableOpacity;
  const MockBackButtonIcon = (props: any) => (
    <TouchableOpacity {...props} testID="back-button" />
  );
  MockBackButtonIcon.displayName = "BackButtonIcon";
  return MockBackButtonIcon;
});

describe("Header", () => {
  it("hides AppTitle when keyboard is open (progress=1)", () => {
    jest.mock("react-native-keyboard-controller", () => ({
      useReanimatedKeyboardAnimation: () => ({ progress: { value: 1 } }),
    }));
    const { toJSON } = render(<Header />);
    // You can check the style or snapshot for the hidden state
    expect(toJSON()).toMatchSnapshot();
  });

  it("shows AppTitle when keyboard is closed (progress=0)", () => {
    const { toJSON } = render(<Header />);
    // You can check the style or snapshot for the visible state
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders BackButtonIcon and AppTitle", () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId("back-button")).toBeTruthy();
  });

  it("calls router.back when BackButtonIcon is pressed", () => {
    const mockBack = jest.fn();
    jest
      .spyOn(require("expo-router"), "useRouter")
      .mockReturnValue({ back: mockBack });
    const { getByTestId } = render(<Header />);
    fireEvent.press(getByTestId("back-button"));
    expect(mockBack).toHaveBeenCalled();
  });

  it("matches the snapshot", () => {
    const { toJSON } = render(<Header />);
    expect(toJSON()).toMatchSnapshot();
  });
});
