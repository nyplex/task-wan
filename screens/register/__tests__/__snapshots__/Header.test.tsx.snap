// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header hides AppTitle when keyboard is open (progress=1) 1`] = `
<View>
  <button
    disabled={false}
    onPress={[Function]}
    testID="back-button"
  />
  <View
    className="overflow-hidden"
    style={
      {
        "height": undefined,
        "marginBottom": undefined,
        "opacity": undefined,
        "transform": [
          {
            "scale": undefined,
          },
        ],
      }
    }
  />
</View>
`;

exports[`Header matches the snapshot 1`] = `
<View>
  <button
    disabled={false}
    onPress={[Function]}
    testID="back-button"
  />
  <View
    className="overflow-hidden"
    style={
      {
        "height": undefined,
        "marginBottom": undefined,
        "opacity": undefined,
        "transform": [
          {
            "scale": undefined,
          },
        ],
      }
    }
  />
</View>
`;

exports[`Header shows AppTitle when keyboard is closed (progress=0) 1`] = `
<View>
  <button
    disabled={false}
    onPress={[Function]}
    testID="back-button"
  />
  <View
    className="overflow-hidden"
    style={
      {
        "height": undefined,
        "marginBottom": undefined,
        "opacity": undefined,
        "transform": [
          {
            "scale": undefined,
          },
        ],
      }
    }
  />
</View>
`;
